core-tests/shared/src/test/scala/zio/ZIOSpec.scala

```diff
@@ -2639,17 +2639,20 @@ object ZIOSpec extends ZIOBaseSpec {
         } yield assert(v)(equalTo(InterruptStatus.uninterruptible))
       },
       testM("executor is heritable") {
-        val io =
-          for {
-            ref  <- Ref.make(Option.empty[internal.Executor])
-            exec = internal.Executor.fromExecutionContext(100)(scala.concurrent.ExecutionContext.Implicits.global)
-            _ <- withLatch(release =>
-                  IO.descriptor.map(_.executor).flatMap(e => ref.set(Some(e)) *> release).fork.lock(exec)
-                )
-            v <- ref.get
-          } yield v.contains(exec)
-
-        assertM(io)(isTrue)
+        val threadName = ZIO.effectTotal(Thread.currentThread.getName)
+        val executor = internal.Executor.fromExecutionContext(100) {
+          scala.concurrent.ExecutionContext.Implicits.global
+        }
+        val poolName =
+          if (TestPlatform.isJS) "main"
+          else if (TestVersion.is211) "ForkJoinPool"
```

Comment:
Did you try using `Thread.currentThread.getThreadGroup`? May be less brittle than substring testing of the thread name.

Reply:
Good idea. Used now.

Reply:
Actually `getThreadGroup` doesn't exist on ScalaJS so we need to revert to using the thread names.